benchmark/Streamly/Benchmark/Data/Parser.hs

```diff
@@ -397,10 +397,16 @@ o_n_heap_serial value =
 
 main :: IO ()
 main = do
-    (value, cfg, benches) <- parseCLIOpts defaultStreamSize
     env <- mkHandleBenchEnv
+#if TASTY
+    (value, ingredients) <- parseCLIOpts defaultStreamSize $ bgroup "All" (allBenchmarks env 0 [])
+    arrays <- IP.toList $ IP.arraysOf 100 $ sourceUnfoldrM value 0  
+    value `seq` defaultMainWithIngredients ingredients $ bgroup "All" (allBenchmarks env value arrays)
+#else
+    (value, cfg, benches) <- parseCLIOpts defaultStreamSize
     arrays <- IP.toList $ IP.arraysOf 100 $ sourceUnfoldrM value 0
     value `seq` runMode (mode cfg) cfg benches (allBenchmarks env value arrays)
+#endif
```

Comment:
Define allBenchmarks as an IO action `allBenchmarks env value =` and call `runWithCLIOpts (allBenchmarks env)`. `arrays` can then be allocated inside allBenchmarks.